docusaurus/docs/ui-extensions/create-an-extension-point.md

```diff
@@ -15,40 +15,288 @@ keywords:
+```
+
+#### Parameters
+
+##### `options.extensionPointId` - _string_
+
+The unique identifier of your extension point. It must begin with `plugins/<PLUGIN_ID>`. For example: `plugins/myorg-super-app`.
+
+##### `options?.context` - _object (Optional)_
+
+An object containing information related to your extension point that you would like to share with the extensions. For example: `{ baseUrl: '/foo/bar' }`. This parameter is not available for component extensions, instead you should pass contextual information using the component props.
+
+:::note
+
+The provided context object always gets frozen (turned immutable) before being shared with the extensions.
+
+:::
+
+##### `options?.limitPerPlugin` - _number (Optional)_
+
+Use this parameter to set the maximum value for how many extensions should be returned from the same plugin. It can be useful in cases when there is limited space on the UI to display extensions.
+
+#### Return value
+
+The hooks return an object in the following format:
+
+```tsx
+// usePluginExtensions()
+{
+  isLoading: boolean;
+  extensions: Array<PluginExtensionLink | PluginExtensionComponent>
+}
+
+// usePluginLinkExtensions()
+{
+  isLoading: boolean;
+  extensions: PluginExtensionLink[]
+}
+
+// usePluginComponentExtensions()
+{
+  isLoading: boolean;
+  extensions: PluginExtensionComponent[]
+}
+```
+
+(For more information check the type definitions of [`PluginExtensionLink`](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/pluginExtensions.ts#L27) and [`PluginExtensionComponent`](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/pluginExtensions.ts#L35C13-L35C37).)
+
+#### Example - rendering link extensions (static context)
+
+The following example shows how to render a link component as link-type extensions that other plugins registered for the `plugins/another-app-plugin/menu` extension point ID.
+
+```tsx
+import { usePluginLinkExtensions } from '@grafana/runtime';
+
+// We define the `context` outside of the React component for performance reasons.
+// (Declaring it inside the component would result in a new object on every render,
+// which would unnecessarily trigger the `usePluginLinkExtensions()` hook.)
+const context = {
+  referenceId: '12345',
+  timeZone: 'UTC',
+};
+
+function AppMenuExtensionPoint() {
+  // This only returns type="link" extensions
+  const { extensions } = usePluginLinkExtensions({
+    extensionPointId: 'plugins/another-app-plugin/menu',
+    context,
+  });
+
+  if (extensions.length === 0) {
+    return null;
+  }
+
+  return (
+    <div>
+      {extensions.map((extension) => {
+        return (
+          <a href={extension.path} onClick={extension.onClick} title={extension.description} key={extension.key}>
+            {extension.title}
+          </a>
+        );
+      })}
+    </div>
+  );
+}
+```
+
+#### Example - rendering link extensions (dynamic context)
```

Comment:
I still find static vs dynamic a little unclear as to why the different situations occur.

Reply:
Good feedback, let's discuss how the best we could explain this difference in the docs.